### PR TITLE
Switch to *.podspec.json format for better automation.

### PR DESCRIPTION
**Pull request by mitchdenny**
As part of #532 we need to have the Azure SDK specs in ```*.podspec.json``` format so that we can run some automation to enforce versioning logic etc. This is the initial step (cutting over to ```*.podspec.json```.

### CODE REVIEW

File: eng/pipelines/templates/steps/verify-versions.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+steps:
+- task: PowerShell@2
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/verify-versions.yml:2]: Why make a template for this as opposed to adding the one step in the other template?

**Review by weshaggard — APPROVED** [review]
I don't have much context on podspec files but the other changes look reasonable.

**Comment by mitchdenny** [comment]
Thanks Wes. I'm just waiting for @bsiegel to be ready to take these changes. I think we are clear of releases now. We need these in master so we can continue the rest of the automation.

**Comment by bsiegel** [comment]
Need to regenerate podspecs based on the latest versions in master.

**Comment by mitchdenny** [comment]
@bsiegel regenerated.

**Review by bsiegel — APPROVED** [review]
This looks good to me. From here we'd like to also get to the point where we can update the versions in the `.jazzy/*.yml` files, the READMEs, and perhaps even the xcodeproj files.

**Pull request merged**